Library/SharedFunctions.swift

```diff
@@ -778,16 +778,20 @@ private func estimatedMinMax(
   var max: Double = 0
 
   rewards.forEach { reward in
-    guard reward.shipping.enabled, let shippingRules = reward.shippingRules else { return }
+    guard reward.shipping.enabled, let shippingRules = reward.shippingRules else {
+      return
+    }
 
     var shippingRule: ShippingRule?
 
-    /// If the reward's shipping prefernce is Anywhere in the world, use its first and only shipping rule.
-    /// Else use the rule that  matches the selected shipping rule (from the locations dropdown).
-    shippingRule = reward.shipping.preference == .unrestricted
-      ? shippingRules.first
-      : shippingRules
-      .first(where: { $0.location.id == locationId })
+    /// First check to see if the reward has a shipping rule that matches the selected location id.
+    /// If it doesn't AND the reward has an unrestricted shipping preference, grab the "Anywhere in the world' shipping rule by id. (id is always 1) .
+    /// Otherwise, we know that the reward doesn't have any shipping rule information associated with it.
+    if let matchingShippingRule = shippingRules.first(where: { $0.location.id == locationId }) {
+      shippingRule = matchingShippingRule
+    } else if reward.shipping.preference == .unrestricted {
+      shippingRule = shippingRules.first(where: { $0.location.id == 1 })
```

Comment:
Once we switch to using `shippingRulesExpanded` we no longer need this line. (And this logic won't always work, which is why we needed to fetch expanded shipping rules for filtering anyways; ex shipping a reward within the EU when there's a shipping rule for all the EU will not show the correct amount.)